packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -463,925 +463,858 @@ describe('NetworkController', () => {
       });
     });
 
-    for (const providerConfig of Object.values(NetworkType).map((networkType) =>
-      buildProviderConfig({ type: networkType }),
-    )) {
-      describe(`when the type in the provider configuration is "${providerConfig.type}"`, () => {
-        describe('if the network ID and network details requests resolve successfully', () => {
-          describe('if the current network is different from the network in state', () => {
-            it('updates the network in state to match', async () => {
-              await withController(
-                {
-                  state: {
-                    networkId: null,
-                    providerConfig,
-                  },
-                },
-                async ({ controller }) => {
-                  await setFakeProvider(controller, {
-                    stubs: [
-                      {
-                        request: { method: 'net_version' },
-                        response: { result: '12345' },
-                      },
-                    ],
-                    stubLookupNetworkWhileSetting: true,
-                  });
-
-                  await controller.lookupNetwork();
-
-                  expect(controller.state.networkId).toBe('12345');
-                },
-              );
-            });
-
-            it("publishes NetworkController:providerConfigChange with the current provider config (even though it didn't change)", async () => {
-              await withController(
-                {
-                  state: {
-                    providerConfig,
-                  },
-                },
-                async ({ controller, messenger }) => {
-                  await setFakeProvider(controller, {
-                    stubs: [
-                      {
-                        request: { method: 'net_version' },
-                        response: { result: '12345' },
-                      },
-                    ],
-                    stubLookupNetworkWhileSetting: true,
-                  });
+    [NetworkType.mainnet, NetworkType.goerli, NetworkType.sepolia].forEach(
+      (networkType) => {
+        describe(`when the provider config in state contains a network type of "${networkType}"`, () => {
+          lookupNetworkTests(
+            buildProviderConfig({ type: networkType }),
+            async (controller: NetworkController) => {
+              await controller.lookupNetwork();
+            },
+          );
+        });
+      },
+    );
 
-                  const providerConfigChanges = await waitForPublishedEvents(
-                    messenger,
-                    'NetworkController:providerConfigChange',
-                    {
-                      produceEvents: async () => {
-                        await controller.lookupNetwork();
-                      },
-                    },
-                  );
+    describe('when the provider config in state contains a network type of "rpc"', () => {
```

Comment:
It looks like this test is the same as the `describe` for the networks list above — did we want to continue including `rpc` in that list or keep it separate?

Reply:
Good point, I can do that. This PR pre-dates the one it's based upon (I started at the last PR and worked backwards to split it up), so I hadn't thought to include the RPC type in the same list by this point :sweat_smile:

Reply:
Done in https://github.com/MetaMask/core/pull/1320/commits/f1b17bd7ab2494a00d5bbea466b7afc165577087